src/main/java/com/faforever/client/fx/WebViewConfigurer.java

```diff
@@ -68,9 +67,26 @@ public void configureWebView(WebView webView) {
       for (int i = 0; i < nodeList.getLength(); i++) {
         Element link = (Element) nodeList.item(i);
         String href = link.getAttribute("href");
-
         link.setAttribute("href", "javascript:java.openUrl('" + href + "');");
       }
+
+      engine.executeScript("""
+          let obs = new MutationObserver((mutations, observer) => {
+            const addedNodes = mutations.flatMap(mut => Array.from(mut.addedNodes));
+            const links = [];
+            for (const node of addedNodes) {
+              if (node?.querySelectorAll) {
+                links.push(...Array.from(document.querySelectorAll("a")));
+              }
+            }
```

Comment:
Only thing besides switching this and the href setting above. But don't we only need to push node.querySelectorAll("a")?

Reply:
I dont think so

It returns a nodelist and pushing that alone didnt seem to explode it correctly

Reply:
node of node list is an array? But you are already running queryselector on it

Reply:
Ill investigate